### PR TITLE
Adaptive Scaffold Widget Package Boilerplating

### PR DESCRIPTION
**Pull request by youssef-attia**
The initial boilerplate for the AdaptiveScaffold suite. First stage in a multistage release. The next stages will introduce actual functionality and examples.

[#108827
](https://github.com/flutter/flutter/issues/108827)
Described further in the [design doc](https://docs.google.com/document/d/1qhrpTWYs5f67X8v32NCCNTRMIjSrVHuaMEFAul-Q_Ms/edit?usp=sharing)

### CODE REVIEW

File: packages/adaptive_helper/example/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+name: example
+description: A new Flutter project.
```
Comment (dkwingsmt) [inline packages/adaptive_helper/example/pubspec.yaml:2]: Better name.

File: packages/adaptive_helper/example/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+# Dependencies specify other packages that your package needs in order to work.
+# To automatically upgrade your package dependencies to the latest versions
+# consider running `flutter pub upgrade --major-versions`. Alternatively,
+# dependencies can be manually updated by changing the version numbers below to
+# the latest version available on pub.dev. To see which dependencies have newer
+# versions available, run `flutter pub outdated`.
+dependencies:
+  adaptive_helper:
+    path: ../../adaptive_helper
```
Comment (dkwingsmt) [inline packages/adaptive_helper/example/pubspec.yaml:33]: Shall we change this (and the package name) to `adaptive_scaffold`?

File: packages/adaptive_helper/example/web/manifest.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+    "name": "example",
+    "short_name": "example",
```
Comment (dkwingsmt) [inline packages/adaptive_helper/example/web/manifest.json:3]: Better name

File: packages/adaptive_helper/example/web/manifest.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "start_url": ".",
+    "display": "standalone",
+    "background_color": "#0175C2",
+    "theme_color": "#0175C2",
+    "description": "A new Flutter project.",
```
Comment (dkwingsmt) [inline packages/adaptive_helper/example/web/manifest.json:8]: Just put anything but the placeholder (I know nobody actually reads it...)

File: packages/adaptive_helper/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+name: adaptive_helper
+description: A new Flutter package project.
```
Comment (dkwingsmt) [inline packages/adaptive_helper/pubspec.yaml]: Name and description

**Review by dkwingsmt — APPROVED** [review]
Some minor change requests

File: packages/adaptive_scaffold/pubspec.yaml [hunk 6]
```diff
@@ -0,0 +1,56 @@
+name: adaptive_scaffold
+description: This package contains some helper widgets that make the process of developing adaptive layouts easier, especially with navigational elements.
+version: 0.0.1
+issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3Aadaptive_scaffold
+repository: https://github.com/flutter/packages/adaptive_scaffold
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:5]: This should be `https://github.com/flutter/packages/tree/main/packages/adaptive_scaffold`.

File: packages/adaptive_scaffold/pubspec.yaml [hunk 7]
```diff
@@ -0,0 +1,56 @@
+name: adaptive_scaffold
+description: This package contains some helper widgets that make the process of developing adaptive layouts easier, especially with navigational elements.
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:2]: Can we make this less verbose? "This package contains" is unnecessary in the description of a package since all package descriptions should be describing what a package contains, "some" doesn't add anything, etc. This is prominently displayed in pub.dev UIs, so should be clear and concise.

How about something more like: "Widgets to easily build adaptive layouts, including navigation elements."

File: packages/adaptive_scaffold/pubspec.yaml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+version: 0.0.1
+issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3Aadaptive_scaffold
+repository: https://github.com/flutter/packages/adaptive_scaffold
+homepage:
+
+environment:
+  sdk: '>=2.17.0 <3.0.0'
+  flutter: ">=1.17.0"
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:10]: Please don't use a minimum version older than 2.8, as that's the earliest we do any automated tests for in CI.

In practice, the Dart SDK version above means you are restricting to Flutter 3.0, so it would be less confusing to just say that.

File: packages/adaptive_scaffold/pubspec.yaml [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+dependencies:
+  flutter:
+    sdk: flutter
+
+dev_dependencies:
+  flutter_lints: ^2.0.0
+  flutter_test:
+    sdk: flutter
+
+# For information on the generic Dart part of this file, see the
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:21]: Please remove all the commented boilerplate from this file.

File: packages/adaptive_scaffold/pubspec.yaml [hunk 10]
```diff
@@ -0,0 +1,56 @@
+name: adaptive_scaffold
+description: This package contains some helper widgets that make the process of developing adaptive layouts easier, especially with navigational elements.
+version: 0.0.1
+issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3Aadaptive_scaffold
+repository: https://github.com/flutter/packages/adaptive_scaffold
+homepage:
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:6]: This should not be here.

File: packages/adaptive_scaffold/lib/adaptive_scaffold.dart [hunk 11]
```diff
@@ -0,0 +1,11 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library adaptive_scaffold;
+
+/// A Calculator.
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/lib/adaptive_scaffold.dart:7]: Why are we checking in a package with no actual functionality? We should wait until we have at least some useful, real content before landing a package (and especially before *publishing* a package, which this PR will do currently.)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/adaptive_scaffold/lib/adaptive_scaffold.dart (hunk 11 above)
Comment (youssef-attia) [reply packages/adaptive_scaffold/lib/adaptive_scaffold.dart:7]: Because of the size of the PR. It was recommended to split it into 3 PRs to make it easier to review. This PR would just hold boilerplate, the next would hold code/tests, and the last would hold the examples. I expect the PRs to be sent out in succession so it would be fully out before anyone could use it in some stable version. Is this not a good way to handle it?

Comment (stuartmorgan) [reply packages/adaptive_scaffold/lib/adaptive_scaffold.dart:7]: If it's going to be landed without any functionality, it should have `publish_to: none` in the pubspec until it's actually a useful package.

This seems like an odd cut-point to me though; there's almost nothing in this PR that requires actual review since it's almost all boilerplate, so I'm not sure how including initial functionality would make reviewing it harder. But if @dkwingsmt feels strongly about landing this now, it needs to not be published.

Comment (youssef-attia) [reply packages/adaptive_scaffold/lib/adaptive_scaffold.dart:7]: I think just having the boilerplate at first makes it easier to drop the perceived changes significantly and makes the review process simpler for everyone involved. I've added the publish_to:none flag but if there's anything else you'd like me to do to make this more streamlined please let me know. Thanks.

File: packages/adaptive_scaffold/pubspec.yaml [hunk 12]
```diff
@@ -0,0 +1,20 @@
+name: adaptive_scaffold
+description: Widgets to easily build adaptive layouts, including navigation elements.
+version: 0.0.1
+issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3Aadaptive_scaffold
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:4]: I missed this before: is this actually the label this package will be using? Usually our package labels are of the form "p: <name of package>". Is there a reason to deviate from that here?

File: packages/adaptive_scaffold/pubspec.yaml [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+repository: https://github.com/flutter/packages/tree/main/packages/adaptive_scaffold
+publish_to: none
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:6]: Please include a comment about why this is here.

File: packages/adaptive_scaffold/pubspec.yaml [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+
+dependencies:
+  flutter:
+    sdk: flutter
+
+dev_dependencies:
+  flutter_lints: ^2.0.0
+  flutter_test:
+    sdk: flutter
+flutter:
```
Comment (stuartmorgan) [inline packages/adaptive_scaffold/pubspec.yaml:20]: Why is this here?

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
Also, please update the PR description to describe the PR, not the project. It's fine to link to the design document for context (although there should also be an issue tracking this, ideally, which you would link to for easier tracking), but the primary description should be of what is in *this PR*, not what the whole project will be.

**Comment by youssef-attia** [comment]
Moved to https://github.com/flutter/packages/pull/2406.
